node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -452,14 +451,22 @@ void tick() {
                 context.log(logger, LogLevel.ERROR, "Unhandled exception, ignoring.", e);
             }
         }
+    }
 
-        doAtTickEnd(converged);
+    private String stateDescription(Node.State state) {
+        return state == null ? "[absent]" : state.toString();
     }
 
     // Public for testing
     void converge() {
         final Optional<NodeSpec> optionalNode = nodeRepository.getOptionalNode(context.hostname().value());
 
+        Node.State newState = optionalNode.map(NodeSpec::getState).orElse(null);
```

Comment:
This can be done a few lines below (478-486) without introducing a new variable

Reply:
the next statement is a return

Reply:
Which is fine as it means the node is removed from node-repo and will shortly be stopped (which is logged).

Reply:
logging moved down